### PR TITLE
[FSDP2] Fix backward-compatible imports

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142419

Internal only: the before way meant that `from torch.distributed._composable.fsdp import fully_shard` was importing `fully_shard.py` not the function `fully_shard`. For some reason, the resolution order is different from open source.

To fix this, we match the old import as closely as possible. Namely, we import `fully_shard.py` contents from `.fully_shard`. This should force that import to take precedence.

cc @H-Huang @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

@diff-train-skip-merge

Differential Revision: [D66990327](https://our.internmc.facebook.com/intern/diff/D66990327)

### CODE REVIEW

**Review by weifengpy — APPROVED** [review]

File: torch/distributed/_composable/fsdp/__init__.py [hunk 1]
```diff
@@ -1,8 +1,3 @@
-from torch.distributed.fsdp import (
```
Comment (awgu) [inline torch/distributed/_composable/fsdp/__init__.py:1]: Internal only: the before way meant that `from torch.distributed._composable.fsdp import fully_shard` was importing `fully_shard.py` not the function `fully_shard`. For some reason, the resolution order is different from open source.

To fix this, we match the old import as closely as possible. Namely, we import `fully_shard.py` contents from `.fully_shard`. This should force that import to take precedence.

**Comment by awgu** [comment]
Since all distributed unit tests in `pytorch` repo are migrated to use the new import, this should be a safe change for the `pytorch` unit tests.

I checked that
```
pytest test/distributed/_composable/fsdp/test_fully_shard_init.py -k test_old
```
passes locally still.

I will have to ninja land this from internal to ensure internal gets this fix asap. I will do so after I verify that lint finishes.

**Comment by awgu** [comment]
Both lintrunner jobs (clang, noclang) have passed. I am going to land.

**Pull request closed without merging**